flamingo/flamingo.h

```diff
@@ -238,29 +274,196 @@ struct flamingo_t {
 	} primitive_type_members[FLAMINGO_VAL_KIND_COUNT];
 };
 
+/**
+ * Raise an error.
+ *
+ * This formats the error message and stores it in the flamingo instance.
+ * If an error is already outstanding, the new error is appended to the existing one.
+ *
+ * @param flamingo The flamingo instance.
+ * @param fmt The format string (printf-style).
+ * @param ... The arguments for the format string.
+ * @return Always returns -1.
+ */
 __attribute__((format(printf, 2, 3))) int flamingo_raise_error(flamingo_t* flamingo, char const* fmt, ...);
 
+/**
+ * Create a new flamingo instance.
+ *
+ * This initializes the flamingo instance and parses the source code.
+ * Note that this does NOT set up the runtime environment (scopes), that happens in {@link flamingo_run} or {@link flamingo_inherit_env}.
+ * Note the `src` buffer is NOT copied.
+ * It must remain valid for the lifetime of the flamingo instance.
+ *
+ * @param flamingo The {@link flamingo_t} instance to initialize.
```

Comment:
these I think are superfluous, as the types are already in the parameters. They were fine as they were before

```suggestion
 * @param flamingo The flamingo instance to initialize.
```